### ISSUE TITLE
rufus w/o UPX - possible?

### ISSUE THREAD

**Issue by eladkarako** [pbatard/rufus#1120]
Any chance of getting a rufus exe,
where both main exe and embedded resources aren't UPX compressed?

**Comment by eladkarako** [comment]
@pbatard I know UPX is great and all,
but my PC at work doesn't like it (no way of disabling the A/V and all the UPX-compressed tools kept deleted as soon as they are identified).
upx -d rufus-2.18p.exe works but the embedded resources are still blocked from executing.

<em>any way of having an anti-virus-friendly version?</em>

I'm in the middle of compiling it from source anyway,
(skipping the UPX part..) but it will take ages on my home-PC using remote-desktop :...(

**Comment by JonnyTech** [comment]
Exactly the same situation here - always having to compile from source too. (Damn corporate anti-malware systems - cannot disable them even with administrator privileges!)

**Comment by pbatard** [comment]
Not gonna happen. I could literally write an essay explaining why this is a waste of time for me and why you have alternatives. Here's the summary:

* First of all, your __likely__ assertion that it's going to be a less than 30 seconds job for me to upload an uncompressed binary for each release, and that therefore it should be trivial for me to do it, is __wrong__. There are tons of consequences from this __seemingly__ innocuous request, such as a lifetime of having to explain to clueless users which of the compressed and uncompressed version is 'better' (because surely if I put 2 versions on the server, one __must__ be better than the other), the required double signing and double uploading, which adds time to a release process that is already lengthy enough and that I am trying to reduce. Uploading stuff doesn't boil down to just an upload. There are hidden costs, especially in time, and my time is in short supply. If you're going to ask me to waste a lot of it on stuff I should never ever have to do, please don't be surprised if you find me fighting back hard.
* If you need Rufus for your job, __YOU__ should be able to make the case for an exception to be added for it with your I.T. department. If you can't do that, then I can guarantee that they're going to find other ways to prevent the download even if uncompressed.
* If they are paranoid to the point of preventing any UPX downloads, they're probably going to prevent Rufus downloading files from its servers, which will break functionality. Next thing you're going to ask me is to work around that... and then work around the next thing they break... and the next thing... See the trend here?
* If you are going to run Rufus, then you have admin rights. This means you should have enough privileges to be able to work around firewall restrictions and stuff on your own.
* If I give an inch to you here, then I'm going to have to give an arm to others there (_"Hey, you helped those guys... So I really don't see why you don't want to move your whole Rufus server to AWS so that I can access them from my filtering corporate environment. It's the same thing!! Why should you help those guys and not meeeee???"_). In other words, what might seem like a no-brained nice thing to do in the short term can end up being detrimental to everyone in the long term. So if I'm going to look like an asshole by having to refuse stuff for seemingly no good reason eventually (at least from the perspective of the requestor), I'd rather cut to the chase and impersonate that asshole right from the start. Saves a lot of __my__ time.
* I've already tried, for years, to fight/work with AV vendors and restrictive corporate environment. And I've reached the conclusion that it was __a complete waste of my time__. Heck, that could be the only point I need to provide here.
* There are `.appx` you can try to use in [testing](https://rufus.akeo.ie/testing/). Either by installing it since you have admin privs (you'll nee to right click every time you launch the app so that it gets admin rights but that's your problem), or, since there's a non UPX exe in there, by uncompressing the stuff and extracting it.
* Likewise, if you know how to do it, it's peanuts to clone the Rufus project and create your appveyor script that will make the `.exe` artefact downloadable.
* Likewise, since you have admin privs, you should be able to install MSYS2/MinGW and compile Rufus from source inside your corporate env. (that is unless your I.T. dept doesn't want you to do that, but if that's the case, then they probably don't want __you__ to use Rufus either, and it's really not my job to fight your I.T. dept for you!).
* This is an Open Source project for crying out loud! If it doesn't provide you with what you want, __YOU__ already have everything you need to modify it so that it does. In this case, producing a vanilla uncompressed `.exe` from the project is as peanut an operation as it gets (and if you want to dispute that statement, then congrats, now you are starting to understand the actual cost of doing stuff __which you can only truly grasp when you aren't asking somebody else to do it for you__).

TL;DR - My time is limited and fighting against AV vendors or corporate firewalls is a waste of my time. Not gonna happen.
